src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -260,6 +237,68 @@ void evaluate(
     }
   }
 }
+}  // namespace detail
+
+/*!
+ * \ingroup TensorExpressionsGroup
+ * \brief Evaluate a RHS tensor expression to a tensor with the LHS index order
+ * set in the template parameters
+ *
+ * \details Uses the right hand side (RHS) TensorExpression's index ordering
+ * (`RhsTE::args_list`) and the desired left hand side (LHS) tensor's index
+ * ordering (`LhsTensorIndices`) to fill the provided LHS Tensor with that LHS
+ * index ordering. This can carry out the evaluation of a RHS tensor expression
+ * to a LHS tensor with the same index ordering, such as \f$L_{ab} = R_{ab}\f$,
+ * or different ordering, such as \f$L_{ba} = R_{ab}\f$.
+ *
+ * The symmetry of the provided LHS Tensor need not match the symmetry
+ * determined from evaluating the RHS TensorExpression according to its order of
+ * operations. This allows one to specify LHS symmetries (via `lhs_tensor`) that
+ * may not be preserved by the RHS expression's order of operations, which
+ * depends on how the expression is written and implemented.
+ *
+ * ### Example usage
+ * Given two rank 2 Tensors `R` and `S` with index order (a, b), add them
+ * together and fill the provided resultant LHS Tensor `L` with index order
+ * (b, a):
+ * \code{.cpp}
+ * TensorExpressions::evaluate<ti_b, ti_a>(
+ *     make_not_null(&L), R(ti_a, ti_b) + S(ti_a, ti_b));
+ * \endcode
+ *
+ * This represents evaluating: \f$L_{ba} = R_{ab} + S_{ab}\f$
+ *
+ * Note: The LHS `Tensor` cannot be part of the RHS expression, e.g.
+ * `evaluate(make_not_null(&L), L() + R());`, because the LHS `Tensor` will
+ * generally not be computed correctly when the RHS `TensorExpression` is split
+ * up and the LHS tensor components are computed by accumulating the result of
+ * subtrees (see the section on splitting in the documentation for the
+ * `TensorExpression` class).
+ *
+ *
+ * Note: `LhsTensorIndices` must be passed by reference because non-type
```

Comment:
Doxygen has a `\note` functionality.

It would also be good to say why this is safe (assuming it is) because dangling references can be a really painful thing to debug :)

Reply:
`\note` addressed in new fixup https://github.com/sxs-collaboration/spectre/pull/3842/commits/904587939fe39bf688e0670b7b21474eada6b315

I'm not sure what you're referring to being safe. If you meant that it should be explained when and why you can use the LHS tensor on the RHS, this is addressed by new fixup https://github.com/sxs-collaboration/spectre/pull/3842/commits/7adc3cd3837d1eac47e914b3a3fd6f967a53ae21

Let me know if you're looking for something else

Reply:
I'm wondering about the `auto&` as the template parameter. I assume it's only used for type info? This comes up because I don't know what the reference points to (it is really a pointer) and so it could be good to clarify that :)

Reply:
I see. Yes, it's only used for type info.

Can you explain what about it being a reference makes you feel it's worth commenting on why it's safe? For example, is there a common dangling pointer pitfall with reference tparams that you're thinking of and concern I should assuage? I only know using references as tparams is something you can do, but I don't know things like this about them.

I'm asking these things just so I know what dangling pointer danger I should be addressing in the comment and explaining why it's not encountered.

Reply:
- [ ] Wait, a guess: are you referring to the possibility of the tparams pointing to a TensorIndex object whose lifetime has ended? If so, then maybe the answer is "we don't have to worry about dangling references because the TensorIndexs you can use are all global static objects" and therefore will never be dangling pointers, unless someone decides to be chaotic and pass their own custom TensorIndex temporaries as tparams. If this unlikely but problematic case of someone using their own custom temporary TensorIndexs is a concern, let me know and I can think about how to prevent people from doing it!

Reply:
Result frrom Google Meet discussion: no need to worry about dangling pointer because the tparams are global static